### ISSUE TITLE
TypeScript declarations for plugin contributors

### ISSUE THREAD

**Issue by jsejcksn** [wulkano/kap#794]
Hi everyone.

I haven't been here in a while, but revisited and thought about contributing a plugin for gfycat support. If I do, I'd like to write it in TypeScript. After a glance at the project, it looks it's written in vanilla JS. Has anyone created a declaration file for [`ShareServiceContext`](https://github.com/wulkano/kap/blob/17bff061650def7041d428bc0b15bd51d68343c4/main/share-service-context.js) by chance?

**Comment by sindresorhus** [comment]
@connor4312 said he would do it soon (https://github.com/connor4312/kap-azure/pull/1#issuecomment-582477582).

**Comment by connor4312** [comment]
It's on my todo. This weekend most likely.

**Comment by jsejcksn** [comment]
@connor4312 Thanks!

@sindresorhus [I finished the first version of the plugin](https://github.com/jsejcksn/kap-gfycat). It works for `gif` and `mp4`. You're welcome to leave a code review. The process has made me realize that the plugin system could use some improvements around messaging. Is there a place where this discussion is taking place or would comments here be best?

**Comment by sindresorhus** [comment]
> Is there a place where this discussion is taking place or would comments here opening a new issue be best?

Open a new issue on this repo. We're happy to discuss :)

**Comment by sindresorhus** [comment]
@connor4312 Just a quick note for when you do. https://github.com/connor4312/kap-azure/blob/f902290eba51395214a4921fe65c09fec11d7602/src/types/kap.d.ts#L23 should be strongly-typed. Either a union or [enum](https://github.com/jsejcksn/kap-gfycat/blob/master/src/index.ts).

**Comment by connor4312** [comment]
Yep, sounds good. I actually ran into an issue when I naively added jpg and a few other formats--turns out that caused an error to be thrown internally within Kap that I had to be debug 😄

**Comment by connor4312** [comment]
PR for this in here 👉  https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42416
